spark/src/test/scala/org/apache/spark/sql/delta/util/JsonUtilsSuite.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (2021) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.delta.util
+
+import scala.util.Random
+
+import org.apache.spark.sql.delta.DeltaLog
+import org.apache.spark.sql.delta.actions.CommitInfo
+import org.apache.spark.sql.delta.stats.DataSize
+import com.fasterxml.jackson.core.StreamReadConstraints
+
+import org.apache.spark.sql.QueryTest
+import org.apache.spark.sql.catalyst.TableIdentifier
+import org.apache.spark.sql.test.SharedSparkSession
+
+class JsonUtilsSuite
+  extends QueryTest
+  with SharedSparkSession {
+
+  test("DataSize json serialization") {
+    val testCases = Seq(
+      DataSize() -> """{}""",
+      DataSize(bytesCompressed = Some(816L)) -> """{"bytesCompressed":816}""",
+      DataSize(rows = Some(111L)) -> """{"rows":111}""",
+      DataSize(logicalRows = Some(111L)) -> """{"logicalRows":111}""",
+      DataSize(bytesCompressed = Some(816L), rows = Some(111L), logicalRows = Some(111L)) ->
+        """{"bytesCompressed":816,"rows":111,"logicalRows":111}"""
+    )
+    for ((obj, json) <- testCases) {
+      assert(JsonUtils.toJson(obj) == json)
+      assert(JsonUtils.fromJson[DataSize](json) == obj)
+    }
+  }
+
+  test("Serialize and de-serialize commit info with large message") {
+    val operationStringSize = StreamReadConstraints.DEFAULT_MAX_STRING_LEN * 10
+    assert(operationStringSize > StreamReadConstraints.DEFAULT_MAX_STRING_LEN)
```

Comment:
Is this a bit redundant, since the variable is defined based on this constant?

Reply:
Did you mean to assert operationStringSize < Int.MaxValue (which can also be its own constant like Delta_JSON_COMMIT_MAX_String_LEN

Reply:
What this is supposed to check is that when the value of DEFAULT_MAX_STRING_LEN changes, the *10 does not lead to an overflow because for this test to be relevant, operationStringSize has to be bigger than the default max (because we want to validate that we have a custom default)